pom.xml

```diff
@@ -301,17 +301,22 @@
     </dependency>
 
     <dependency>
-      <groupId>log4j</groupId>
-      <artifactId>log4j</artifactId>
-      <version>1.2.17</version>
-      <scope>runtime</scope>
+      <groupId>org.slf4j</groupId>
+      <artifactId>slf4j-api</artifactId>
+      <version>1.7.26</version>
     </dependency>
 
     <dependency>
       <groupId>junit</groupId>
       <artifactId>junit</artifactId>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>ch.qos.logback</groupId>
+      <artifactId>logback-classic</artifactId>
```

Comment:
You saw that this is a just a test dependency?
I'm rather reluctant to turn this into an IT. As you can see, I have several tests and I will need more of those in the future because #201 would also need to check the log output.
If I turned all of them into ITs, it would take too much time for me to run them.
I could try to convert my library to work with the simple logging framework implementation used my Maven itself but I had the code and this solution was easiest for me.

Reply:
Note: I'm trying to get this into the slf4j code base: https://jira.qos.ch/browse/SLF4J-461

Reply:
I spent a couple of hours to write a very simple logger implementation which would also make the tests pass. The code is here: https://github.com/digulla/license-maven-plugin/commit/5eb32342cd1bd608aec98a9bc18cba303fe6d1a9
It quite a few lines of code. On the plus side, we can now add additional checks (= number of {} matches number of arguments).

Reply:
Yes, I saw, this is a test dependency and that `org.codehaus.mojo.license.logback.*` is test code and I still think it is too much code for what it does. I'd prefer to stop checking the log output for the cases we have now. The methods under test would be tested well enough by checking their return values or thrown exceptions.

If you insist on some way to verify logged messages in the future, I prefer discussing it separately at the point in time when we start needing it. 

This PR contains a lot of useful changes and I would not like to block it because of this.

Reply:
Maven communicates with users by the text we print on the console. In other software, logging is used to do post mortem debugging. But even there, I'd argue you want to include important log messages in the tests to make sure they are logged and that they contain useful information.
That's why I argue the things we log are part of the API contract. It's part of the "return values" which you want me to test.
Therefore, I want to include those in the tests. I did that before and it wasn't a problem.

Many people don't know how to test this. That's why there is a feeling "not possible" / "don't try this".
I do this in all my projects. The code I added is a simplified copy of my own log testing library. The code was written two years ago and I had no bugs, just a couple of feature requests in that time.

There are several issues in the license-maven-plugin bugtracker which ask to make changes to the output. Without the framework, we won't be able to test those changes in unit tests.
I don't like ITs. They are slow, brittle, take a long time to set up, hard to debug, and when they break, you have to spend a lot of time figuring out what they are supposed to do. To add insult to injury, each of the ITs in the project take 30-120 seconds to run on Windows.

As for "we don't need it now": Those tests need it now. They already existed. They checked log output before and if I removed that, I would cripple existing tests. I wanted to avoid that.

Therefore, I think I have several good arguments for my approach.

What exactly is the problem? Is it too many lines of code per PR? Or are you afraid that some change will break the code and no one will be able to fix it since it's too complicated?

Reply:
Yes, I agree ITs are often an overkill (and perhaps also an overkill for this particular case) At the same time I dare to insist that adding the logback dependency and 300+ lines of new logback code is an overkill too. I am not doubting the quality of the code, I have concerns about the future maintainability of the project. Less code and less dependencies are better. Logback and five new classes are too much for what we need.

As I said, I'd be ready to sacrifice the log verification altogether and you are right that I do not consider log messages to be a part of the method contract. I'd say if the messages are to be seen as part of the contract, they should become part of the directly testable method output. Either passing a `org.slf4j.Logger` as a method param or some simplified easily mockable interface would be IMO much cheaper than logback + 300 LoC. I'd still prefer to split this discussion from the present PR, so that other potentially conflicting work can go on.

Reply:
I'm sorry that I can't do this with less code. I strongly believe that the log messages should be checked which is why I invented a framework to be able to do it in a comfortable and safe way.
The slf4j API isn't very well suited for this, which is why I need that much code.
I also don't want to cripple existing tests and I'm out of time.
I'm really trying here to help but I feel that my expertise isn't appreciated. I'm doing this to help and to scratch an itch that I have and at the moment, it's not fun at all. If you don't accept this PR either, I've wasted some 30 hours which I could have put into a fork that fixes my problems. So for me, this isn't really working out and I have a feeling it's not working for you either.